attrs 21.3.0 breaks coverage report on Python 3.6 (regression from 21.2.0)

With attrs 21.3.0, if I run a script that does `import attr` using `coverage`, I get an error from `coverage report`.

```console
$ python3 --version
Python 3.6.15
$ echo 'import attr' > test.py
$ pip3 install attrs coverage
…
Successfully installed attrs-21.3.0 coverage-6.2
$ coverage run test.py
$ coverage report
No source for code: '/tmp/rep-1>'.
```

This worked with attrs 21.2.0.

```console
$ pip3 install attrs==21.2.0
…
Successfully installed attrs-21.2.0
$ coverage run test.py
$ coverage report
Name      Stmts   Miss  Cover
-----------------------------
test.py       1      0   100%
-----------------------------
TOTAL         1      0   100%
```

I bisected the problem to commit 5a368d82997570bdad2ba371e897cc777809461f (#858). Cc @thetorpedodog @hynek.

python 3.7 has same issue

working in CI/CD

```
Running with gitlab-runner 14.6.0~beta.71.gf035ecbf (f035ecbf)
  on green-5.shared.runners-manager.gitlab.com/default xS6Vzpvo
Preparing the "docker+machine" executor
Using Docker executor with image python:3.7-slim-buster ...
Pulling docker image python:3.7-slim-buster ...

[...]

Collecting attrs>=19.2.0
  Downloading attrs-21.3.0-py2.py3-none-any.whl (61 kB)
[...]

coverage report -i
/usr/local/lib/python3.7/site-packages/coverage/report.py:87: CoverageWarning: Couldn't parse '/builds/Jade/jade/jadeapp/jade-rest-auth/rep-10>': No source for code: '/builds/Jade/jade/jadeapp/jade-rest-auth/rep-10>'. (couldnt-parse)
  coverage._warn(msg, slug="couldnt-parse")
/usr/local/lib/python3.7/site-packages/coverage/report.py:87: CoverageWarning: Couldn't parse '/builds/Jade/jade/jadeapp/jade-rest-auth/rep-11>': No source for code: '/builds/Jade/jade/jadeapp/jade-rest-auth/rep-11>'. (couldnt-parse)
  coverage._warn(msg, slug="couldnt-parse")
/usr/local/lib/python3.7/site-packages/coverage/report.py:87: CoverageWarning: Couldn't parse '/builds/Jade/jade/jadeapp/jade-rest-auth/rep-12>': No source for code: '/builds/Jade/jade/jadeapp/jade-rest-auth/rep-12>'. (couldnt-parse)
  coverage._warn(msg, slug="couldnt-parse")
/usr/local/lib/python3.7/site-packages/coverage/report.py:87: CoverageWarning: Couldn't parse '/builds/Jade/jade/jadeapp/jade-rest-auth/rep-13>': No source for code: '/builds/Jade/jade/jadeapp/jade-rest-auth/rep-13>'. (couldnt-parse)

[...]

coverage xml
No source for code: '/builds/DeepNatural/deepnaturalai/deepapi/deepapi-rest-auth/rep-10>'.
make: *** [Makefile:37: coverage] Error 1
```

but when i fixed attrs==21.2.0, its work

--- a/changelog.d/895.change.rst
+++ b/changelog.d/895.change.rst
@@ -1,0 +1,1 @@
+Fix ``coverage report`` for projects who use ``attrs`` and don't set a ``--source``.

--- a/changelog.d/896.change.rst
+++ b/changelog.d/896.change.rst
@@ -1,0 +1,1 @@
+Fix ``coverage report`` for projects who use ``attrs`` and don't set a ``--source``.

--- a/src/attr/_make.py
+++ b/src/attr/_make.py
@@ -1888,7 +1888,7 @@ def _add_eq(cls, attrs=None):
 if HAS_F_STRINGS:
 
     def _make_repr(attrs, ns, cls):
-        unique_filename = "repr"
+        unique_filename = _generate_unique_filename(cls, "repr")
         # Figure out which attributes to include, and which function to use to
         # format them. The a.repr value can be either bool or a custom
         # callable.

